src/main/java/de/tum/cit/aet/artemis/exercise/repository/StudentParticipationRepository.java

```diff
@@ -1361,4 +1361,48 @@ SELECT MAX(pr.id)
                 ORDER BY p.student.firstName ASC
             """)
     Page<FeedbackAffectedStudentDTO> findAffectedStudentsByFeedbackId(@Param("exerciseId") long exerciseId, @Param("feedbackIds") List<Long> feedbackIds, Pageable pageable);
+
+    /**
+     * Retrieves the logins of students affected by a specific feedback detail text in a given exercise.
+     *
+     * @param exerciseId The ID of the exercise for which affected students are requested.
+     * @param detailText The feedback detail text to filter by.
+     * @return A list of student logins affected by the given feedback detail text in the specified exercise.
+     */
+    @Query("""
+                SELECT DISTINCT p.student.login
+                FROM ProgrammingExerciseStudentParticipation p
+                LEFT JOIN p.submissions s
+                LEFT JOIN s.results r
+                LEFT JOIN r.feedbacks f
+                WHERE p.exercise.id = :exerciseId
+                  AND f.detailText = :detailText
+                  AND p.testRun = FALSE
+                  AND f.positive = FALSE
+            """)
+    List<String> findAffectedLoginsByFeedbackDetailText(@Param("exerciseId") long exerciseId, @Param("detailText") String detailText);
+
+    /**
+     * Counts the number of distinct students affected by a specific feedback detail text for a given programming exercise.
+     * <p>
+     * This query identifies students whose submissions were impacted by feedback entries matching the provided detail text
+     * within the specified exercise. Only students with non-test run submissions and negative feedback entries are considered.
+     * </p>
+     *
+     * @param exerciseId the ID of the programming exercise for which the count is calculated.
+     * @param detailText the feedback detail text used to filter the affected students.
+     * @return the total number of distinct students affected by the feedback detail text.
+     */
+    @Query("""
+                SELECT COUNT(DISTINCT p.student.id)
+                FROM ProgrammingExerciseStudentParticipation p
+                LEFT JOIN p.submissions s
+                LEFT JOIN s.results r
+                LEFT JOIN r.feedbacks f
```

Comment:
Why are we using LEFT JOIN here? Because `exerciseId` cannot be null and `detailText`should not be null. So, I think an inner join should be more efficient here.